tests/db-utils.ts

```diff
@@ -123,12 +123,35 @@ export async function cleanupDb(prisma: PrismaClient) {
 	try {
 		// Disable FK constraints to avoid relation conflicts during deletion
 		await prisma.$executeRawUnsafe(`PRAGMA foreign_keys = OFF`)
-		await prisma.$transaction([
-			// Delete all rows from each table, preserving table structures
-			...tables.map(({ name }) =>
-				prisma.$executeRawUnsafe(`DELETE from "${name}"`),
-			),
-		])
+
+		// Delete tables except the ones that are excluded above
+		for (const { name } of tables) {
+			await prisma.$executeRawUnsafe(`DROP TABLE IF EXISTS "${name}"`)
+		}
+
+		const migrationPaths = fs
+			.readdirSync('prisma/migrations')
+			.filter((dir) => dir !== 'migration_lock.toml')
+			.map((dir) => `prisma/migrations/${dir}/migration.sql`)
+
+		// Run each migration
+		for (const path of migrationPaths) {
+			const sql = fs.readFileSync(path, 'utf8')
+			const statements = sql
+				.split(';')
+				.map((statement) => statement.trim())
+				.filter(Boolean)
+
+			// Run each sql statement in the migration
+			for (const statement of statements) {
+				try {
+					await prisma.$executeRawUnsafe(`${statement};`)
+				} catch (error) {
+					console.warn(`Failed to execute statement: ${statement}`, error)
+					// Continue with the next statement
```

Comment:
We should probably re-throw the error here actually, otherwise we'd have a partially migrated database which could be very bad/confusing.

```suggestion
					console.warn(`Failed to execute statement: ${statement}`)
					throw error
```